EOLib/Domain/Login/PlayerInfoRepository.cs

```diff
@@ -10,8 +10,6 @@ public interface IPlayerInfoRepository
 
         short PlayerID { get; set; }
```

Comment:
Since you refactored the welcome packets to use the same pattern as everything else with session IDs, I think you can remove the `PlayerID` property here entirely. The only purpose of having it here was to pass it between the welcome request / welcome message handlers. I see you're using it now for the connection ID but I don't think that's used anywhere outside of the initial connection

Reply:
But this is where player ID is set. The ID sent from welcome reply is just another session ID so you kind of need to keep it here and set it from the connection action :)

Reply:
Is there somewhere else player ID is stored for other in game actions that require it?

Reply:
I'm not seeing it used for any other in-game actions - everything else uses `Character.ID`.

I think it actually has been working by accident up until now. In welcome agree, I take the character matching the logged in character's name from the list of map characters and use that as the "main character". This has the side effect of overwriting the existing character ID (which is just character select ID) with the player ID, since this is what is sent from eoserv.

Reply:
It's kind of weird. The only place GameServer sends you your PlayerID is on the initial handshake.

But it does expect you to send back the CharacterID on Welcome_Msg
```
"Entering game"
client_packet(Welcome, Msg)
{
	three session_id
	int character_id
}
```

From what I can see in the protocol CharacterID is never used after entering the game. Looks like the server only expects it when
- Requesting login to a character
- deleting a character

So maybe there's no point of tracking CharacterID outside of the "LoggedIn" controller and we always just set the MainCharacter.ID to PlayerID from connecting? I guess we'd still have to track it in the PlayerInfoRepository right?

Reply:
I went ahead and just left PlayerID in the repository and overwrite MainCharacter's ID with that when completing login

Reply:
I think I'm going to clean this up further once it's merged. Right now `mainCharacter` is matched on the logged in character's name, it would be cleaner to match on the ID instead